app/queries/consumer_apps/all_query.rb

```diff
@@ -0,0 +1,14 @@
+module ConsumerApps
+  class AllQuery
```

Comment:
should this be called `FindOrCreateAllQuery` ?

Reply:
I agree this would be a better name for it. I'll be pushing the changes in a bit